src/api-types.ts

```diff
@@ -793,9 +844,23 @@ export interface PeoplePropertyValue extends PropertyValueBase {
   people: User[]
 }
 
+type FileWithName = {
+  name: string
+} & (File | ExternalFile)
+
+// When writing, we don't allow S3 hosted files
+type ExternalFileWithName = {
+  name: string
+} & ExternalFile
+
 export interface FilesPropertyValue extends PropertyValueBase {
   type: "files"
-  files: { name: string }[]
+  files: FileWithName[]
+}
+
+export interface FilesPropertyInputValue extends PropertyValueBase {
+  type: "files"
+  files: ExternalFileWithName[]
```

Comment:
tiny nit: the `"When writing, we don't allow S3 hosted files"` would make more sense above `FilesPropertyInputValue` because it explains why this is the input.